.github/workflows/build_and_test.yaml

```diff
@@ -11,7 +11,34 @@ on:
 env:
   GO_VERSION: 1.18.2
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          version: v1.45.0
```

Comment:
We'll need to keep an eye on this, as golangci-lint doesn't yet support Go generics stuff, so any generics we add may cause this check to fail.

Reply:
I think there may be a more recent version available as well - we should go with the latest version we can find.

Reply:
@youngnick 
Agree, referring to the [Contour golangci-lint](https://github.com/projectcontour/contour/blob/main/.github/workflows/prbuild.yaml#L20), it does not set up golang environment either. And yes, we'll need to keep an eye on this.

Thanks for your reminder, the latest version is [1.46.2](https://github.com/golangci/golangci-lint/releases/tag/v1.46.2) for now. Will bump the version.
xref: [golangci-lint/releases](https://github.com/golangci/golangci-lint/releases)